### ISSUE TITLE
add lmod module system to handle sub-containers?

### ISSUE THREAD

**Issue by stebo85** [NeuroDesk/vnm#12]

**Comment by MartinGrignard** [comment]
Sounds a good solution to manage multiple versions of softwares.

**Comment by MartinGrignard** [comment]
Strange, I'm trying to add lmod to the image. When I use:
```shell
RUN apt-get update \
    && apt-get install --no-install-recommends -y \
        lua-bit32 \
        lua-filesystem \
        lua-json \
        lua-lpeg \
        lua-posix \
        lua-term \
        lua5.2 \
        tcl \
        tcl-dev \
        libtcl8.6 \
        lmod \
    && rm -rf /var/lib/apt/lists/*
```
the build works but no command is added to the final image so issuing `module list` or any other command, I get `module: command not found`...

**Comment by thomshaw92** [comment]
I found this https://github.com/readthedocs/readthedocs-docker-images/compare/master...shahzebsiddiqui:master 
Try adding
RUN ln -s /usr/lib/x86_64-linux-gnu/lua/5.2/posix_c.so /usr/lib/x86_64-linux-gnu/lua/5.2/posix.so

@MartinGrignard

**Comment by MartinGrignard** [comment]
Yep, that's what I did in the last update :) It works when installing from inside the session but not at the build stage. Not quiet sure why yet... If you have any clue about why the lmod.sh script is not created, I'm interested :D

**Comment by stebo85** [comment]
Hi @MartinGrignard - I have an idea whats wrong :) you need to source the module system in every shell you are using - Here is how we set this up in our puppet scripts on our workstations: Would you like to give it a go to try that in docker? PM me for more input :)

# installs module system
class environment_module_system {
  case $operatingsystem {
    'Ubuntu','Debian': {
      package { 'lmod':
        ensure          => latest,
        install_options => ['--allow-unauthenticated', '-f']
      }
      file {'/etc/environment-modules':
        ensure => directory
      }
      ~> file {'/etc/environment-modules/modulespath':
              ensure => present
      }
      ~> file_line { 'cvl_module_path':
        ensure => present,
        line   => '/data/lfs2/CVL/containers/modules/',
        path   => '/etc/environment-modules/modulespath',
      }
      ~> file_line { 'std_module_path_del':
        ensure => absent,
        line   => '/usr/share/modules/modulefiles                          # General module files ',
        path   => '/etc/environment-modules/modulespath',
      }
      file { '/etc/profile.d/modules.sh':
          source => '/mnt/cai-it/mgtfiles/PuppetPackages/modules.sh',
      }
      ~> file_line { 'source_modules':
        path => '/etc/profile',
        line => 'source /etc/profile.d/modules.sh'
      }
      ~> file_line { 'modulepath':
        path => '/etc/profile',
        line => 'export MODULEPATH=/data/lfs2/CVL/containers/modules'
      }



case $operatingsystemrelease {
        '18.04' : {
            file { '/usr/lib/x86_64-linux-gnu/lua/5.1/posix.so':
                ensure => 'link',
                target => '/usr/lib/x86_64-linux-gnu/lua/5.1/posix_c.so',
            }
                                        file { '/usr/lib/x86_64-linux-gnu/lua/5.2/posix.so':
                                                ensure => 'link',
                                                target => '/usr/lib/x86_64-linux-gnu/lua/5.2/posix_c.so',
                                        }
                                        file { '/usr/lib/x86_64-linux-gnu/lua/5.3/posix.so':
                                                ensure => 'link',
                                                target => '/usr/lib/x86_64-linux-gnu/lua/5.3/posix_c.so',
                                        }
                                }
                        }
    }
  }
}



This is in modules.sh
#----------------------------------------------------------------------#
# system-wide profile.modules                                          #
# Initialize modules for all sh-derivative shells                      #
#----------------------------------------------------------------------#
trap "" 1 2 3

case "$0" in
    -bash|bash|*/bash) . /usr/share/lmod/6.6/init/bash ;;
       -ksh|ksh|*/ksh) . /usr/share/lmod/6.6/init/ksh ;;
       -zsh|zsh|*/zsh) . /usr/share/lmod/6.6/init/zsh ;;
          -sh|sh|*/sh) . /usr/share/lmod/6.6/init/sh ;;
                    *) . /usr/share/lmod/6.6/init/sh ;;  # default for scripts
esac

trap - 1 2 3
~

**Comment by MartinGrignard** [comment]
@stebo85 I'll give it a try right now :)

**Comment by MartinGrignard** [comment]
Closed in #18.
